crates/sui-core/src/authority/authority_store_pruner.rs

```diff
@@ -189,53 +104,51 @@ impl AuthorityStorePruner {
             if effects.iter().any(|effect| effect.is_none()) {
                 return Err(anyhow::anyhow!("transaction effects data is missing"));
             }
-            Self::handle_checkpoint(effects.into_iter().flatten(), &perpetual_db.objects)?;
-            checkpoint_store.update_highest_pruned_checkpoint(&checkpoint)?;
-            pruned_seq_number += 1;
+            batch_effects.extend(effects.into_iter().flatten());
+
```

Comment:
If we have `prune_effects` return `HashMap<ObjectID, (VersionNumber, VersionNumber)>` we won't need to batch effects in memory and can do a bulk rocksdb delete. Right now we are bound to invoke delete across fewer effects since we probably can't keep many of those in memory